File: web-api/src/models/regulator/settings_model.py
from models.abstracts.app_base_model import AppBaseModel

from models.regulator.control_parameters_model import ControlParametersModel
from models.regulator.enums.regulator_state_model import RegulatorStateModel
from models.regulator.regulator_owner_model import RegulatorOwnerModel
from models.regulator.service_model import ServiceModel
from models.common.signin_model import SigninModel


class SettingsModel(AppBaseModel):
    mhenoscheme_name: str

    regulator_state: RegulatorStateModel

    signin: SigninModel

    regulator_owner: RegulatorOwnerModel

    control_parameters: ControlParametersModel

    service: ServiceModel

File: web-ui/src/models/regulator-settings/settings-model.ts
import { SigninModel } from '../data/signin-model'
import { ControlParametersModel } from './control-parameters-model'
import { RegulatorStateModel } from './enums/regulator-state-model'
import { GpioSetModel } from './gpio-set-model'
import { RegulatorOwnerModel } from './regulator-owner-model'
import { ServiceModel } from './service-model'


export type RegulatorSettings = {
    mhenoschemeName: string,

    regulatorState: RegulatorStateModel,

    signIn: SigninModel,

    regulatorOwner: RegulatorOwnerModel,

    controlParameters: ControlParametersModel,

    service: ServiceModel,

    gpioSet: GpioSetModel
}